interop/xds/server/server.go

```diff
@@ -78,8 +83,47 @@ func (s *testServiceImpl) EmptyCall(ctx context.Context, _ *testpb.Empty) (*test
 }
 
 func (s *testServiceImpl) UnaryCall(ctx context.Context, in *testpb.SimpleRequest) (*testpb.SimpleResponse, error) {
+	response := &testpb.SimpleResponse{ServerId: s.serverID}
+
+	for headerValue := range getRPCBehaviorMetadata(ctx) {
+		if rpcStatus := getStatusForRPCBehaviorMetadata(headerValue); rpcStatus != nil {
+			return response, status.Err(rpcStatus.Code(), rpcStatus.Message())
+		}
+	}
+
+	response.Hostname = s.hostname
```

Comment:
Just set hostname on line 86 inline in the struct.

Reply:
@zasweq I did it purposely because even in the cpp implementation as well they aren't assigning it when a status is found for the RPC behaviour: https://github.com/grpc/grpc/blob/be99673d069217def9036cc9ae0346da324ba373/test/cpp/interop/xds_interop_server.cc#L122

Reply:
would you still like me to move it?

Reply:
I don't think the c++ implementation is right. I'd say go ahead and move it.

Reply:
Commented on the c++ implementation.